core/src/test/java/com/amplifyframework/core/model/query/predicate/PredicateTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.core.model.query.predicate;
+
+import com.amplifyframework.testmodels.personcar.Person;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+
+import static com.amplifyframework.core.model.query.predicate.QueryPredicateOperation.not;
+
+/**
+ * Tests predicate creation.
+ */
+@SuppressWarnings("MagicNumber")
+public final class PredicateTest {
+    /**
+     * Tests the creation of a simple single operation predicate.
+     */
+    @Test
+    public void testSingleQueryPredicateOperation() {
+        QueryPredicateOperation<?> op = Person.ID.eq("1234");
+
+        assert (op.field().equals("id"));
+        assert (op.operator().getClass().equals(EqualQueryOperator.class));
+        assert (((EqualQueryOperator) op.operator()).value().equals("1234"));
```

Comment:
Should use jUnit's `assert<Whatever>(...)` family of methods instead of these `assert`s.

[Some discussion around this is provided on StackOverflow](https://stackoverflow.com/questions/2966347/assert-vs-junit-assertions).

Reply:
This part of the test was written by David, so I didn't touch it (simply moved the file + added additional tests)